tests/e2e/util.go

```diff
@@ -5570,3 +5570,51 @@ func waitForPvcToBeDeleted(ctx context.Context, client clientset.Interface, pvcN
 	framework.Logf("Status of pvc is: %v", pvc.Status.Phase)
 	return waitErr
 }
+
+/* This util method fetches pvc events list and matches the events list with the
+specified error message and returns true if expected error found */
+func getEventsMessageListAndVerifyPvcError(client clientset.Interface, namespace string,
+	pvclaim *v1.PersistentVolumeClaim, expectedErrMsg string) (bool, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	isFailureFound := false
+	ginkgo.By("Checking for error in events related to pvc " + pvclaim.Name)
+	waitErr := wait.PollImmediate(poll, pollTimeoutShort, func() (bool, error) {
+		eventList, _ := client.CoreV1().Events(namespace).List(ctx,
+			metav1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%s", pvclaim.Name)})
+		for _, item := range eventList.Items {
+			if strings.Contains(item.Message, expectedErrMsg) {
+				framework.Logf("Expected Error msg found. EventList Reason: "+
+					"%q"+" EventList item: %q", item.Reason, item.Message)
+				isFailureFound = true
+				break
+			}
+		}
+		return isFailureFound, nil
+	})
+	return isFailureFound, waitErr
+}
+
+/* This util method fetches pvc events list and matches the events list with the
+specified error message and returns true if expected error found */
+func getEventsListAndVerifyPvcError(client clientset.Interface, namespace string,
+	pvclaim *v1.PersistentVolumeClaim, expectedErrMsg string) (bool, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	isFailureFound := false
+	ginkgo.By("Checking for error in events related to pvc " + pvclaim.Name)
+	waitErr := wait.PollImmediate(poll, pollTimeoutShort, func() (bool, error) {
+		eventList, _ := client.CoreV1().Events(namespace).List(ctx,
+			metav1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%s", pvclaim.Name)})
+		for _, item := range eventList.Items {
+			if strings.Contains(item.Reason, expectedErrMsg) {
+				framework.Logf("Expected Error msg found. EventList Reason: "+
+					"%q"+" EventList item: %q", item.Reason, item.Message)
+				isFailureFound = true
+				break
+			}
+		}
+		return isFailureFound, nil
+	})
+	return isFailureFound, waitErr
```

Comment:
The code here looks dupe of the function getEventsMessageListAndVerifyPvcError - you can avoid duplicate code and parameterize the dynamic validation.